core-rust/state-manager/src/state_manager.rs

```diff
@@ -264,21 +297,66 @@ impl StateManager {
         // ... and for deleting the old, non-critical ledger proofs (a.k.a. "Proofs GC"):
         let ledger_proofs_gc = LedgerProofsGc::new(
             database.clone(),
-            config.ledger_proofs_gc_config,
-            config.ledger_sync_limits_config,
+            config.ledger_proofs_gc_config.clone(),
+            config.ledger_sync_limits_config.clone(),
         );
         scheduler
             .named("ledger_proofs_gc")
             .start_periodic(ledger_proofs_gc.interval(), move || ledger_proofs_gc.run());
 
         Self {
+            config,
             state_computer,
             database,
             pending_transaction_result_cache,
             mempool,
             mempool_manager,
+            execution_configurator,
             committability_validator,
             transaction_previewer,
+            protocol_updater_factory: protocol_updater_factory.into(),
         }
     }
+
+    pub fn apply_protocol_update(&self, protocol_version_name: &str) -> ProtocolUpdateResult {
```

Comment:
Is there a reason why this must be called from the outside? It seems like protocol updates should be in the realm of StateManager. But this is minor and we can refactor this later.